share-coordinator/src/main/java/org/apache/kafka/coordinator/share/PersisterOffsetsStateBatch.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.coordinator.share;
+
+import org.apache.kafka.common.message.WriteShareGroupStateRequestData;
+import org.apache.kafka.coordinator.share.generated.ShareSnapshotValue;
+import org.apache.kafka.coordinator.share.generated.ShareUpdateValue;
+import org.apache.kafka.server.group.share.PersisterStateBatch;
+
+import java.util.Objects;
+
+/**
+ * This is a helper class which overrides the equals and hashcode
+ * methods to only focus on the first and last offset fields of the
+ * state batch. This is useful when combining batches.
+ */
+public class PersisterOffsetsStateBatch extends PersisterStateBatch {
```

Comment:
Using the type hierarchy for this is a bit unintuative and likely to cause surprises down the road. Imagine you have a `PersisterStateBatch` reference but it's actually a `PersisterOffsetsStateBatch` object and the behavior is different. 

A better approach is to create a wrapper class that has the custom equalsTo and hashCode. In other words, let this class hold a PersisterStateBatch rather than extending that type.

Reply:
will use the delegate